source/dotnet/Library/AdaptiveCards.Rendering.Wpf/Helpers/ImageExtensions.cs

```diff
@@ -125,7 +138,16 @@ public static async void SetBackgroundSource(this Grid grid, Uri uri, AdaptiveRe
                 return;
             }
 
-            BitmapImage bi = await context.ResolveImageSource(finalUri);
+            BitmapImage bi = null;
+            if (finalUri.Scheme == "data")
+            {
+                bi = GetBitmapFromBase64(finalUri);
+            }
+            else
+            {
+                bi = await context.ResolveImageSource(finalUri);
+            }
+
```

Comment:
You probably want to invoke the resource resolver in the case of DataURIs in case the host has registered their own.